### PR TITLE
arrows: update cursor only when in Select mode

### PR DESCRIPTION
**Pull request by mimecuvalo**
The cursor was updating for the arrow label even when in other tools (e.g. Draw) and it should only be updating when in Select mode. That addresses this issue: https://github.com/tldraw/tldraw/issues/2750

Also, there was a problem with arrows and zero length arrows and labels. The problem was actually in `Vec.ts` where we were dividing by zero. Addresses this bug https://github.com/tldraw/tldraw/issues/2749

### Change Type

- [x] `patch` — Bug fix

### Test Plan

1. Make sure that the cursor is only applicable to the Select tool.

### Release Notes

- Cursor tweak for arrow labels.

### CODE REVIEW

File: packages/tldraw/src/lib/tools/SelectTool/childStates/Idle.ts [hunk 1]
```diff
@@ -34,6 +34,28 @@ export class Idle extends StateNode {
 
 	override onPointerMove: TLEventHandlers['onPointerMove'] = () => {
 		updateHoveredId(this.editor)
+
+		const hitShape = this.editor.getShapeAtPoint(this.editor.inputs.currentPagePoint, {
+			hitInside: false,
+			hitLabels: false,
+			margin: HIT_TEST_MARGIN / this.editor.getZoomLevel(),
+			renderingOnly: true,
+		})
```
Comment (ds300) [inline packages/tldraw/src/lib/tools/SelectTool/childStates/Idle.ts:43]: I think you could use this?
```suggestion
		const hitShape = editor.getHoveredShape()
```

File: packages/tldraw/src/lib/tools/SelectTool/childStates/Idle.ts [hunk 2]
```diff
@@ -499,6 +510,28 @@ export class Idle extends StateNode {
...
+
+		// todo: Extract into general hit test for arrows
+		if (this.editor.isShapeOfType<TLArrowShape>(shape, 'arrow')) {
+			// How should we handle multiple labels? Do shapes ever have multiple labels?
+			const labelGeometry = this.editor.getShapeGeometry<Group2d>(shape).children[1]
+			// Knowing what we know about arrows... if the shape has no text in its label,
+			// then the label geometry should not be there.
+			if (labelGeometry && pointInPolygon(pointInShapeSpace, labelGeometry.vertices)) {
+				return true
+			}
```
Comment (ds300) [inline packages/tldraw/src/lib/tools/SelectTool/childStates/Idle.ts:529]: nonblocking: I know this was in a previous PR, but it feels brittle to me, i wonder whether we could put the logic on the ArrowShapeUtil class, something like `arrowUtil.isPointWithinArrowLabelBounds(arrow, pointInShapeSpace)`?

File: packages/tldraw/src/lib/tools/SelectTool/childStates/Idle.ts [hunk 3]
```diff
@@ -34,6 +34,28 @@ export class Idle extends StateNode {
...
+			margin: HIT_TEST_MARGIN / this.editor.getZoomLevel(),
+			renderingOnly: true,
+		})
+		if (this.isOverArrowLabelTest(hitShape)) {
+			this.editor.updateInstanceState({
+				cursor: {
+					type: 'pointer',
+					rotation: 0,
+				},
+			})
```
Comment (SomeHats) [inline packages/tldraw/src/lib/tools/SelectTool/childStates/Idle.ts:50]: this is fine, but you could also use `this.editor.setCursor` for these if you wanted

File: packages/editor/src/lib/primitives/Vec.ts [hunk 4]
```diff
@@ -362,7 +362,10 @@ export class Vec {
 	 * Get the unit vector of A.
 	 */
 	static Uni(A: VecLike) {
-		return Vec.Div(A, Vec.Len(A))
+		const len = Vec.Len(A)
+		if (len === 0) return Vec.From(A)
+
+		return Vec.Div(A, len)
```
Comment (SomeHats) [inline packages/editor/src/lib/primitives/Vec.ts:368]: discussed offline - moving this out of `Vec` to the callsite instead

**Review by SomeHats — APPROVED** [review]

File: packages/tldraw/src/lib/tools/SelectTool/childStates/Idle.ts (hunk 2 above)
Comment (mimecuvalo) [reply packages/tldraw/src/lib/tools/SelectTool/childStates/Idle.ts:529]: sounds good - will followup later 👍

**Pull request merged**